### DISCUSSION TITLE
How do I set environment variables when running an application using WSLG

### DISCUSSION THREAD

**Discussion by sunny-zhang-98** [microsoft/wslg#995] [Q&A]
How do I set environment variables when running an application using WSLG, I want set proxy variable befor wslg app running

**Comment by Biswa96** [comment]
Environment variables can be set as usual. There should be no difference if wslg is running or not.

**Reply by sunny-zhang-98** [reply]
But when I use wslg the environment variables are not in effect, I set them in /etc/profile

**Reply by gnaggnoyil** [reply]
AFAIK in most distributions GUI applications doesn't read configurations in `/etc/profile`. You could try modifying that application's `.desktop` file(either by modifying its content directly, or if that desktop file lies in `/usr/share/applications`, cpoying that desktop file to `$XDG_DATA_HOME/applications` then doing the modifications on that copied file), or just run your application in your shell session.
